src/Pages/Nodes.razor

```diff
@@ -308,6 +351,7 @@
         {
             if (node != null)
             {
+                await StopJob(node);
                 var (result, _) = NodeRepository.Remove(node);
                 if (!result)
```

Comment:
If the node cannot be removed, shouldn't the job be restored?

Reply:
But if the user doesn't get to delete the node it leaves a state where a node is in the system but the jobs for it are not running. It doesn't sound good to me.

Reply:
So we need to stop the job before deletion the node.